ComfortZoneApp/app/src/main/java/com/example/comfortzone/utils/ComfortCalcUtil.java

```diff
@@ -62,26 +70,99 @@ private static int getSum(LevelsTracker tracker) {
 
     public static void calculateAverages(ParseUser currentUser) {
         ArrayList<LevelsTracker> trackerArrayList = (ArrayList<LevelsTracker>) currentUser.get(KEY_LEVEL_TRACKERS);
-        List<Task<Void>> savingAverages = trackerArrayList.stream().filter(tracker -> tracker.getCount() > 0).map(tracker -> calculateSingularAverage(tracker)).collect(Collectors.toList());
+        List<Task<Void>> savingAverages = trackerArrayList.stream().map(tracker -> calculateSingularAverage(tracker)).collect(Collectors.toList());
         Task.whenAll(savingAverages).onSuccess(new Continuation<Void, Object>() {
             @Override
             public Object then(Task<Void> task) throws Exception {
+                goThroughTrackerList(trackerArrayList);
                 return null;
             }
         });
     }
 
     private static Task<Void> calculateSingularAverage(LevelsTracker tracker) {
-        ArrayList<ComfortLevelEntry> entryArrayList = (ArrayList<ComfortLevelEntry>) tracker.get(KEY_ENTRIESLIST);
+        ArrayList<ComfortLevelEntry> entryArrayList = null;
+        try {
+            entryArrayList = (ArrayList<ComfortLevelEntry>) tracker.fetchIfNeeded().get(KEY_ENTRIESLIST);
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
         int sum = entryArrayList.stream().mapToInt(entry -> entry.getTemp()).sum();
         int count = tracker.getCount();
-        tracker.put(KEY_AVERAGE, sum / count);
+        if (count > 0) {
+            tracker.put(KEY_AVERAGE, sum / count);
+        } else {
+            tracker.put(KEY_AVERAGE, MIN_TEMP);
+        }
         return tracker.saveInBackground();
     }
-    
 
+    private static void goThroughTrackerList(List<LevelsTracker> trackerList) {
+        if (trackerList.get(0).getAverage() < trackerList.get(1).getAverage() || trackerList.get(0).getCount() > trackerList.get(1).getCount()) {
```

Comment:
Can format this if condition to have the second condition on next line like 

if (trackerList.get(0).getAverage() < trackerList.get(1).getAverage() 
     || trackerList.get(0).getCount() > trackerList.get(1).getCount()) {